app/DAO/BakeryDatabase.scala

```diff
@@ -105,6 +105,47 @@ class BakeryDatabase @Inject() (
       .transact(bakeryTransactor.xa)
       .unsafeRunSync()
   }
+
+  def updateProduct(id: String, changesJson: JsValue): Int = {
+
+    /** Steps for updating a record
+      *  1. bring in the json, json should be the values to change:
+      *     - name, qty, price, name+qty, name+price, qty+price, name+qty+price
+      *  2. check what values are given to update, form query on these params
+      *  3. query with .update on the given id
+      *  4. run the transaction
+      */
+    val newName = (changesJson \ "name").asOpt[String]
+    val newQty = (changesJson \ "quantity").asOpt[Int]
+    val newPrice = (changesJson \ "price").asOpt[Double]
+    val newStamp = OffsetDateTime.now()
+    var query = sql""""""
```

Comment:
Both pattern matching expressions and if statements return values. Usage of vars is frowned upon. Please remove and use vals